plugins/modules/clustering/consul/consul.py

```diff
@@ -475,6 +492,16 @@ def __init__(self, check_id, name, node=None, host='localhost',
 
             self.check = consul.Check.http(http, self.interval, self.timeout)
 
+        if tcp:
+            if interval is None:
+                raise Exception('tcp check must specify interval')
+            
+            if ':' not in tcp:
+                raise Exception('tcp check must contain port')
+
+            tcpParsedList = tcp.split(':')
+            self.check = consul.Check.tcp(tcpParsedList[0], int(tcpParsedList[1]), self.interval)
```

Comment:
This won't work if the hostname is an IPv6 address.

Reply:
Rewrote using regex